### PR TITLE
Upgrade to ocamlformat.0.20.0

### PR DESCRIPTION
**Pull request by gpetiot**
This is a preview of the not-yet-released `ocamlformat.0.20.0`, please wait until the package is published in opam to merge this PR. The output is still likely to slightly change before the package is released.
The changes are due to:
- `module-item-spacing` is now set to `compact` for the default profile
- `module-item-spacing` is now correctly applied to mutually recursive type definitions
- formatting of list elements has been improved

### CODE REVIEW

File: bench/irmin-pack/trace_stat_summary_pp.ml [hunk 1]
```diff
@@ -796,44 +796,38 @@ module Table4 = struct
         (`S, "Wall time elapsed *C", zip (fun s -> s.elapsed_wall_over_blocks));
       `Data (`S, "CPU time elapsed *C", zip (fun s -> s.elapsed_cpu_over_blocks));
       `Data (`P, "CPU Usage *LA", zip (fun s -> s.cpu_usage.evolution));
-      (* ops counts *)
-      `Spacer;
+      (* ops counts *) `Spacer;
```
[the remark this answers is not in the document]
Comment (CraigFe) [reply bench/irmin-pack/trace_stat_summary_pp.ml]: Fixed by https://github.com/ocaml-ppx/ocamlformat/pull/1876.

**Review by CraigFe — APPROVED** [review]

**Comment by CraigFe** [comment]
The CI-installed `ocamlformat` is trying to introduce extended let-op punning:

```diff
--- a/src/irmin-graphql/server.ml
+++ b/src/irmin-graphql/.formatted/server.ml
@@ -172,7 +172,7 @@ struct
               >>= Lwt.return_some
           | None -> Lwt.return_none
         in
-        let+ parents = parents in
+        let+ parents in
         ( Config.info ?author "%s" message,
           input.retries,
           input.allow_empty,
```

but my understanding is that this should be disabled by setting `ocaml-version = 4.08.0". Not sure what's going on here.

**Comment by gpetiot** [comment]
> The CI-installed `ocamlformat` is trying to introduce extended let-op punning:
> but my understanding is that this should be disabled by setting `ocaml-version = 4.08.0`. Not sure what's going on here.

Since running ocamlformat alone on this code produces the expected formatting I would say that's an  `ocaml-ci` issue. There are always been a bit of lag between a new ocamlformat release and the lint-fmt job passing everywhere.

**Comment by CraigFe** [comment]
The issue here turned out to be that the following is broken:

```
version = 0.20.0
ocaml-version = 4.08.0
profile = conventional
```

Setting `profile` after `ocaml-version` causes the default for `ocaml-version` to be re-applied, ignoring the manually-set version. 

@gpetiot: perhaps OCamlformat should warn when a config value is overwritten by `profile`?

**Comment by gpetiot** [comment]
> @gpetiot: perhaps OCamlformat should warn when a config value is overwritten by `profile`?

I agree and I'm taking note, but unfortunately we would need a significant redesign of our options to be able to do that.

**Pull request merged**